src/textual/widgets/_option_list.py

```diff
@@ -0,0 +1,904 @@
+
+        Note that the size of this list will at least the same as the number
+        of options, and actually greater if any prompt of any option is
+        multiple lines.
+        """
+
+        self._spans: list[OptionLineSpan] = []
+        """A list of the locations and sizes of all options in the option list.
+
+        This will be the same size as the number of prompts; each entry in
+        the list contains the line offset of the start of the prompt, and
+        the count of the lines in the prompt.
+        """
+
+        # Initial calculation of the content tracking.
+        self._request_content_tracking_refresh()
+
+        self._mouse_hovering_over: int | None = None
+        """Used to track what the mouse is hovering over."""
+
+        # Finally, cause the highlighted property to settle down based on
+        # the state of the option list in regard to its available options.
+        # Be sure to have a look at validate_highlighted.
+        self.highlighted = None
+
+    def _request_content_tracking_refresh(
+        self, rescroll_to_highlight: bool = False
+    ) -> None:
+        """Request that the content tracking information gets refreshed.
+
+        Args:
+            rescroll_to_highlight: Should the widget ensure the highlight is visible?
+
+        Calling this method sets a flag to say the refresh should happen,
+        and books the refresh call in for the next idle moment.
+        """
+        self._needs_refresh_content_tracking = True
+        self._needs_to_scroll_to_highlight = rescroll_to_highlight
+        self.check_idle()
+
+    def on_idle(self) -> None:
+        """Perform content tracking data refresh when idle."""
+        self._refresh_content_tracking()
+        if self._needs_to_scroll_to_highlight:
+            self._needs_to_scroll_to_highlight = False
+            self.scroll_to_highlight()
+
+    def watch_show_vertical_scrollbar(self, old: bool, new: bool) -> None:
+        """Handle the vertical scrollbar visibility status changing.
+
+        Args:
+            old: The old status.
+            new: The new status.
+
+        `show_vertical_scrollbar` is watched because it has an impact on the
+        available width in which to render the renderables that make up the
+        options in the list. If a vertical scrollbar appears or disappears
+        we need to recalculate all the lines that make up the list.
+        """
+        if old != new:
+            self._request_content_tracking_refresh()
+
+    def on_resize(self) -> None:
+        """Refresh the layout of the renderables in the list when resized."""
+        self._request_content_tracking_refresh(rescroll_to_highlight=True)
+
+    def on_mouse_move(self, event: MouseMove) -> None:
+        """React to the mouse moving.
+
+        Args:
+            event: The mouse movement event.
+        """
+        self._mouse_hovering_over = event.style.meta.get("option")
```

Comment:
I still don't get what this `style.meta` is all about.
How does it always have all the information you need..?

Reply:
I don't think the meta stuff is very well documented yet (or at all). Might be worth having a chat with Will.